controllers/nodes/nodes_controller.go

```diff
@@ -37,21 +43,103 @@ var (
 	}
 )
 
+// startupNodeLabeler is a runnable that labels all nodes after the manager cache starts
+type startupNodeLabeler struct {
+	reconciler *ReconcileNodeCounter
+}
+
+// Start implements manager.Runnable
+func (s *startupNodeLabeler) Start(ctx context.Context) error {
+	log.Info("Starting node labeling after cache is ready")
+
+	const (
+		maxRetries    = 5
+		retryInterval = 10 * time.Second
+	)
+
+	// Label all nodes now that the cache is ready
+	err := s.reconciler.labelAllNodesAtStartup(ctx)
+	if err != nil {
+		log.Error(err, "Failed to label nodes at startup, will retry", "maxRetries", maxRetries, "retryInterval", retryInterval)
+
+		// Retry up to maxRetries times
+		for attempt := 1; attempt <= maxRetries; attempt++ {
+			select {
+			case <-ctx.Done():
+				log.Info("Context cancelled during retry, stopping node labeling")
+				return nil
+			case <-time.After(retryInterval):
+				log.Info("Retrying node labeling at startup", "attempt", attempt, "maxRetries", maxRetries)
+				if err = s.reconciler.labelAllNodesAtStartup(ctx); err == nil {
+					log.Info("Successfully labeled nodes at startup after retry", "attempt", attempt)
+					break
```

Comment:
I think will break from the case but not from the loop. See here: https://go.dev/ref/spec#Break_statements